src/relations.jl

```diff
@@ -145,6 +146,94 @@ function air_density(
     return p / (gas_constant_air(param_set, q) * T)
 end
 
+"""
+    air_density_equil(
```

Comment:
Why do we need this? That is, why can't we just get the density from temperature, specific humidities, and whatever other thermodynamic variable (energy, entropy etc.) we are using?

Reply:
It seems that we need this in `saturation_adjustment_psq` because temperature is not yet known. I tried using `air_density(..., PhasePartition(q_tot))`, but this leads to consistency failures in the following tests:

```julia
ts_psq = PhaseEquil_psq.(param_set, p, s, q_tot)
@test all(entropy.(ts_psq) .≈ s)                                     # requires air_density_equil
@test all(isapprox.(air_pressure.(ts_psq), p; rtol = rtol_pressure)) # requires air_density_equil
```

Since this is only used for the reference state in TC.jl, I don't think performance is much of an issue. I think this function (`air_density_equil`) could fix consistency errors in other thermo functions, but we're already accurate to the tolerances in the test suite, so I don't think it's much of an issue.

The reason I went down this rabbit hole was because, IIRC, without `air_density_equil`, the TC.jl reference states are slightly different and one or two of the simulations crashed.

Reply:
We can reformulate this without another iterative method. But better yet, we can do without any of it. No need to entropy-based reference states.